src/state.rs

```diff
@@ -71,34 +72,35 @@ impl fmt::Display for Upstream {
     }
 }
 
-/// The current state information for this proxy.
-#[derive(serde::Serialize, Default, Debug)]
-pub struct ProxyState {
-    #[serde(flatten)]
-    pub workloads: WorkloadStore,
-
+/// A Dns Resolver is responsible for the DNS resolving task for given hostnames
+#[derive(serde::Serialize, Default, Debug, Clone)]
+struct DnsResolver {
     #[serde(flatten)]
-    pub services: ServiceStore,
+    resolved_dns: Arc<RwLock<ResolvedDnsStore>>,
 
-    #[serde(flatten)]
-    pub policies: PolicyStore,
+    #[serde(skip_serializing)]
+    dns_resolver_cfg: ResolverConfig,
 
-    #[serde(flatten)]
-    pub resolved_dns: ResolvedDnsStore,
+    #[serde(skip_serializing)]
+    dns_resolver_opts: ResolverOpts,
 }
 
 /// A ResolvedDnsStore encapsulates all resolved DNS information for workloads in the mesh
-#[derive(serde::Serialize, Default, Debug)]
-pub struct ResolvedDnsStore {
+#[derive(serde::Serialize, Default, Debug, Clone)]
```

Comment:
why Clone?

Reply:
Hi @nmittler, as the definition of `resolved_dns: Arc<RwLock<ResolvedDnsStore>>`, so `ResolvedDnsStore` must declare the default clone to support concurrency safe accessing.